### PR TITLE
Add toleration to the API

### PR DESCRIPTION
**Pull request by zsedem**
https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/

Signed-off-by: Adam Oliver Zsigmond <azsigmond@hiya.com>

### CODE REVIEW

File: client/src/main/scala/skuber/Toleration.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package skuber
```
Comment (doriordan) [inline client/src/main/scala/skuber/Toleration.scala]: It is a matter of style really but as tolerations are tied to pods (and used in pod specs only) - I would prefer to define the Toleration type in the Pod scope (i.e. within the Pod companion object) similar to e.g. Pod.Affinity, Pod.Condition. 
Does this seem reasonable?

Comment (zsedem) [reply client/src/main/scala/skuber/Toleration.scala]: I guess it is, just Pod is kinda big already.

**Comment by zsedem** [comment]
Is there anything blocking this now?

**Comment by doriordan** [comment]
Nothing blocking this @zsedem I am just catching up with it now, looks good thanks

**Pull request merged**